### PR TITLE
Circle CI: enable BUILD_HERMES_SOURCE for test_ios_rntester job 

### PR DESCRIPTION
**Pull request by matinzd**
## Summary

Fix CircleCI build for test_ios_rntester.

Broken due to [this commit](https://github.com/facebook/react-native/commit/fefa7b6ac8a1e0e33fa7a1070936c5c83c873c0a) after adding dev tools profiler.

## Changelog

[Internal] [Fixed] - Fix rn tester app CI

## Test Plan
CI should be green.

### CODE REVIEW

**Comment by cortinico** [comment]
Thanks for doing this @matinzd 👍 Let me take a closer look but we should be good merging this if the CI is green

**Comment by matinzd** [comment]
> Thanks for doing this @matinzd 👍 Let me take a closer look but we should be good merging this if the CI is green

I converted it to a draft since there are some problems that need to be fixed. I will let you know when it's ready.

File: .circleci/config.yml [hunk 1]
```diff
@@ -727,6 +727,10 @@ jobs:
           name: Set USE_HERMES=1
           command: echo "export USE_HERMES=1" >> $BASH_ENV
 
+      - run:
+          name: Set BUILD_HERMES_SOURCE=1
+          command: echo "export BUILD_HERMES_SOURCE=1" >> $BASH_ENV
```
Comment (cortinico) [inline .circleci/config.yml:732]: This shouldn't be necessary.

File: scripts/react_native_pods.rb [hunk 2]
```diff
@@ -673,18 +673,18 @@ def downloadAndConfigureHermesSource(react_native_path)
...
     # If hermesc is present, create a symbolic link in the hermes source directory to avoid re-building hermesc
     Pod::UI.puts "[Hermes] Using pre-compiled Hermes Compiler from #{hermesc_macos_path}"
     system("ln -s #{hermesc_macos_path} #{hermesc_macos_link}")
   end
-
-  # TODO: Integrate this temporary hermes-engine.podspec into the actual one located in facebook/hermes
-  system("cp #{sdks_dir}/hermes-engine.podspec #{hermes_dir}/hermes-engine.podspec")
+  
+  Pod::UI.puts "[Hermes] Extracting Hermes (#{hermes_tag_sha})"
+  system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
```
Comment (cortinico) [inline scripts/react_native_pods.rb:687]: Please add:

```
system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
``` 

at line 676.

And don't remove the copy of the `.podspec` file.

File: .circleci/config.yml (hunk 1 above)
Comment (matinzd) [reply .circleci/config.yml:732]: If we do not enable the build Hermes from the source, the newest changes with newly added methods for the profiler will not work. Are you sure?

File: scripts/react_native_pods.rb (hunk 2 above)
Comment (matinzd) [reply scripts/react_native_pods.rb]: Fixed under [00570d9](https://github.com/facebook/react-native/pull/33585/commits/00570d94d2cb4e86609091ac61d12ea03a81887d).

File: .circleci/config.yml (hunk 1 above)
Comment (matinzd) [reply .circleci/config.yml:732]: https://github.com/facebook/react-native/commit/fefa7b6ac8a1e0e33fa7a1070936c5c83c873c0a

**Pull request closed without merging**